backend/databases/sql.py

```
@@ -0,0 +1,216 @@
+    ''')
+    
+    conn.commit()
+    conn.close()
+
+def authenticate_user(username: str, password: str):
+    """Authenticate user - CONTAINS DELIBERATE SQL INJECTION"""
+    conn = sqlite3.connect(DATABASE_PATH)
+    cursor = conn.cursor()
+    
+    # DELIBERATE SQL INJECTION VULNERABILITY
+    query = f"SELECT * FROM users WHERE username = '{username}'"
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)